BigBaseV2/src/memory/range.cpp

```diff
@@ -29,39 +29,81 @@ namespace memory
 		return h.as<std::uintptr_t>() >= begin().as<std::uintptr_t>() && h.as<std::uintptr_t>() <= end().as<std::uintptr_t>();
 	}
 
-	static bool pattern_matches(std::uint8_t* target, const std::optional<std::uint8_t>* sig, std::size_t length)
+
+	//https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore%E2%80%93Horspool_algorithm
+	//https://www.youtube.com/watch?v=AuZUeshhy-s
+	static handle scan_pattern(const std::optional<std::uint8_t>* sig, std::size_t length, handle begin, std::size_t moduleSize)
 	{
-		for (std::size_t i = 0; i < length; ++i)
+		std::size_t maxShift = length;
+		std::size_t maxIdx = length - 1;
+		//Get wildcard index, and store max shifable byte count
+		std::size_t wildCardIdx{ size_t(-1) };
+		for (int i{ int(maxIdx - 1) }; i >= 0; --i)
 		{
-			if (sig[i] && *sig[i] != target[i])
-				return false;
+			if (!sig[i])
+			{
+				maxShift = maxIdx - i;
+				wildCardIdx = i;
+				break;
+			}
 		}
-
-		return true;
+		//Store max shiftable bytes for non wildcards.
+		std::size_t shiftTable[UINT8_MAX + 1]{};
+		for (std::size_t i{}; i <= UINT8_MAX; ++i)
+		{
+			shiftTable[i] = maxShift;
+		}
+		for (std::size_t i{ wildCardIdx + 1 }; i != maxIdx; ++i)
+		{
+			shiftTable[*sig[i]] = maxIdx - i;
+		}
+		//Loop data
+		for (std::size_t curIdx{}; curIdx != moduleSize - length;)
+		{
+			for (std::size_t sigIdx = maxIdx; sigIdx >= 0; --sigIdx)
+			{
+				if (sig[sigIdx] && *begin.add(curIdx + sigIdx).as<uint8_t*>() != *sig[sigIdx])
+				{
+					curIdx += shiftTable[*begin.add(curIdx + maxIdx).as<uint8_t*>()];
+					break;
+				}
+				else if (sigIdx == NULL)
+				{
+					return begin.add(curIdx);
+				}
+			}
+		}
+		return nullptr;
```

Comment:
Variables do not follow coding style.

Reply:
Will commit soon, waiting for review to see if there was anything else.